CDAP-8305 Remove extensions from documentation

awholegunch opened the pull request:
This removes the two CDAP extensions (Hydrator and Tracker) and makes them part of the Developers' Manual (Hydrator as "Pipelines") and the Administration Manual (Tracker as "Metadata Management").

Running as a Quick Build: http://builds.cask.co/browse/CDAP-DQB256-16

Pages of interest:
- Developers' Manual http://builds.cask.co/artifact/CDAP-DQB256/shared/build-16/Docs-HTML/4.1.0-SNAPSHOT/en/developers-manual/index.html
- Metadata http://builds.cask.co/artifact/CDAP-DQB256/shared/build-16/Docs-HTML/4.1.0-SNAPSHOT/en/developers-manual/metadata/index.html
- Pipelines http://builds.cask.co/artifact/CDAP-DQB256/shared/build-16/Docs-HTML/4.1.0-SNAPSHOT/en/developers-manual/pipelines/index.html

Fix for https://issues.cask.co/browse/CDAP-8305

Commit:
Remove extensions, and replace hydrator and tracker with pipelines
and metadata management: work-in-progress

Commit:
w-i-p

Commit:
w-i-p

Commit:
work-in-progress

Commit:
Fix typo.

Commit:
Remove tracker

Commit:
Fix bad links.

Commit:
Move pipelines and metadata-management into developers' guide and
adminisation manual respectively.

Commit:
Remove manuals improve error handling, improve messages.

awholegunch commented:
On-hold because Product has a request.

Commit:
Moves and changes per comments.

bdmogal commented on cdap-docs/_common/_source/index.rst:
```diff
@@ -90,12 +96,26 @@ CDAP extensions provide additional capabilities, such as *Cask Hydrator* and *Ca
   - |dev-man-o-black|_ The overall architecture, abstractions, modes, and components behind CDAP
   - |dev-man-bb-black|_ The two core abstractions in CDAP: *Data* and *Applications*, and their components
+  - |dev-man-m-black|_ CDAP can automatically capture metadata and let you see **how data is flowing**
+  - |dev-man-p-black|_ A capability of CDAP that enables the **building, deploying, and managing of data pipelines**
```
...that allows users to build, deploy and manage...

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/admin-manual/source/installation/ambari.rst:
```diff
@@ -158,7 +158,7 @@ Assign CDAP Services to Hosts
    #. **Auth Server:** For managing authentication tokens on CDAP clusters with perimeter security enabled
    #. **Kafka Server:** For transporting CDAP metrics and CDAP system service log data
-   #. **UI:** Web interface to CDAP and :ref:`Cask Hydrator <cask-hydrator>`
+   #. **UI:** Web interface to CDAP and :ref:`CDAP Application Studio <cdap-application-studio>`
```
lets just call this CDAP Studio.

awholegunch replied:
I've scanned the doc set and replaced this everywhere.

bdmogal commented on cdap-docs/admin-manual/source/operations/cdap-ui.rst:
```diff
@@ -43,8 +43,7 @@ feature parity with the original UI, and is still under development.
 datapacks, and drivers; the Resource Centre, which provides wizards for adding streams,
-drivers, applications, Hydrator pipelines, and plugins; and access to CDAP extensions
-including Cask Hydrator, Cask Tracker, and Cask Wrangler.
+drivers, applications, pipelines, and plugins; and access to CDAP extensions.
```
lets remove extensions

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/admin-manual/source/operations/cdap-ui.rst:
```diff
@@ -43,8 +43,7 @@ feature parity with the original UI, and is still under development.
 
 It includes access to: the Cask Market, with use cases, pipelines, applications, plugins,
 datapacks, and drivers; the Resource Centre, which provides wizards for adding streams,
```
lets remove Resource center - its just called the Add Entity functionality

awholegunch replied:
Removed.

bdmogal commented on cdap-docs/developers-manual/source/_includes/plugins/index.rst:
```diff
@@ -38,7 +38,7 @@ SDK and Distributed CDAP:
 .. rubric:: Plugin Notes
 
-- Plugins are grouped here based on how they are displayed in the Hydrator Studio UI.
+- Plugins are grouped here based on how they are displayed in the CDAP Application Studio UI.
```
CDAP Studio

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/developers-manual/source/_includes/plugins/index.rst:
```diff
@@ -55,7 +55,7 @@ SDK and Distributed CDAP:
 - Certain plugins |---| such as the *JavaScript*, *Python Evaluator*, and *Validator*
   transforms |---| are designed to be customized by end-users with their own code, either
-  from within Hydrator Studio or in a configuration file. For instance, you can create your
+  from within CDAP Application Studio or in a configuration file. For instance, you can create your
```
ditto

awholegunch replied:
ibid.

bdmogal commented on cdap-docs/developers-manual/source/_includes/plugins/index.rst:
```diff
@@ -87,9 +87,9 @@ using:
 If you are creating a custom plugin that extends the **existing system artifacts,** its
-name should not collide with existing names, for ease-of-use in the CDAP UI and Cask
-Hydrator Studio. You are free to create your own plugin and plugin-type, depending on the
+name should not collide with existing names, for ease-of-use in the CDAP UI and CDAP Application
```
ditto

awholegunch replied:
fixed.

bdmogal commented on cdap-docs/developers-manual/source/metadata/index.rst:
```diff
@@ -1,23 +1,31 @@
+.. toctree::
+   :maxdepth: 1
+   
+    CDAP Metadata UI <metadata-ui>
```
This anchor is wrong? It should be overview? Because of this, the CDAP Metadata UI appears as the first link in the table of contents for this section, and seems wrong

awholegunch replied:
Let's discuss how to handle this, as we have run into a current limitation of Sphinx. What's happening is that all rst files need to appear at least once in an table of contents somewhere, so that they can be found and linked together. They are either in a defined `table-of-contents.rst` or in another file, always as part of a ``.. toctree::`` directive. The problem is that the directive always gets inserted first into the sidebar. I don't have control over it, though I guess I could do something about it if it was really required. It would necessitate digging into the Sphinx internals and figuring out how to change the order.

How I have handled it is to instead do a top-level page that has *no* section breaks in it, that is an "overview" page, and then have all pages below it appear as links in the left sidebar.

See http://builds.cask.co/artifact/CDAP-DQB256/shared/build-8/Docs-HTML/4.1.0-SNAPSHOT/en/developers-manual/data-exploration/index.html for an example.

bdmogal commented on cdap-docs/developers-manual/source/metadata/index.rst:
```diff
@@ -1,23 +1,31 @@
-Metadata and Lineage are a new and important feature of CDAP. CDAP Metadata helps show how
-datasets and programs are related to each other and helps in understanding the impact of a
-change before the change is made. 
+*Metadata* (and with *lineage*, a closely-related feature) are important capabilites of
```
lets remove the lineage reference here

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/developers-manual/source/pipelines/concepts-design.rst:
```diff
@@ -78,7 +78,7 @@ Logical and Physical Pipelines
 Within CDAP, there is the concept of *logical* and *physical* pipelines.
 
-A **logical pipeline** is the view as seen in Hydrator Studio and the Cask Hydrator UI.
+A **logical pipeline** is the view as seen in the CDAP Application Studio.
```
ditto

awholegunch replied:
Fixed.

bdmogal commented on cdap-docs/developers-manual/source/pipelines/creating-pipelines.rst:
```diff
@@ -34,7 +34,7 @@ which are used to create the different kinds of data pipeline applications.
 been deprecated and replaced by the above artifacts, as of CDAP 3.5.0.*
 
-Pipelines can be created using Cask Hydrator's included visual editor (*Cask Hydrator
+Pipelines can be created using CDAP's included visual editor (the *CDAP Application
```
ditto

awholegunch replied:
Fixed.

bdmogal reviewed (changes requested):
@awholegunch this looks good overall. Two main issues:
1. CDAP Application Studio -> CDAP Studio
2. The metadata table of contents does not look right - UI appears on top, for example. The other problem is that Sub sections of overview appear at the same level as overview.

Commit:
CDAP Application Studio -> CDAP Studio, plus other changes.

Commit:
Replace "cdap-application-studio" with "cdap-studio"

Commit:
Fix duplicate anchor.

Commit:
Minor edit.

Commit:
Add links to audit logging.

Commit:
Add redirect.

Commit:
Add redirect that handles any anchors.

Commit:
Another link that needs redirecting.

Commit:
Changes per comments.

bdmogal commented:
@awholegunch this is a good start, we can work on this incrementally. Just a few things I noticed:
1. In the metadata page, the sections look mis-aligned. The section titles seem to be more indented than the content of the sections. Perhaps the content should be indented too:
![image](https://cloud.githubusercontent.com/assets/3895099/22958378/bd6450b0-f2e3-11e6-8297-e2f4f80d8205.png)
2. Discover and Lineage
Do these things belong together? Maybe its fine, but it should be **Discovery** and Lineage
3. Can the "Audit Logging" section move before the UI? The flow seems to me that there are certain features of the platform - metadata, system metadata, discovery, lineage, audit logging; and the UI then is a visual indicator of these features:
![image](https://cloud.githubusercontent.com/assets/3895099/22958434/1b714a00-f2e4-11e6-9ba6-745905bec5f4.png)
4. This section in discovery should also say that you can find a dataset or a stream that has a "field with the given name" or a "field with the given name and the given type". I think that is incredibly useful:
![image](https://cloud.githubusercontent.com/assets/3895099/22958490/735f4122-f2e4-11e6-93e9-9fbadc62947d.png)
5. Should we move the Metadata Management REST APIs under the Administering Metadata UI, primarily because it makes references to datasets, etc that are described in the Administering UI section.


Other than these points, this looks good!

Commit:
Fixes per comments.

awholegunch commented:
1. I have removed the rubrics and replaced them with a bullet list.
2. Fixed by editing text as "Discovery".
3. Fixed.
4. That API is very long, and added to the CDAP Metadata UI page would make it very long, which is why it was broken and removed from there originally. I think all RESTful APIs should, as a matter on convention, regardless of topic, be in the ref manual together.

Commit:
Edits.

bdmogal commented on cdap-docs/developers-manual/source/metadata/discovery-lineage.rst:
```diff
@@ -15,15 +15,16 @@ Metadata can be used to tag different CDAP components so that they are easily id
 For example, you can tag a dataset as *experimental* or an application as *production*. These
-entities can then be discovered by using the annotated metadata in **search queries**. Using search,
-you can discover entities:
+entities can then be discovered by using the annotated metadata in **search queries**. 
```
...using search queries...?

awholegunch replied:
Edited.

bdmogal reviewed (approved):
small comment. Otherwise LGTM

Commit:
Edits ore comment.
